rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalTransferServer.java

```diff
@@ -0,0 +1,191 @@
+import java.nio.channels.FileChannel;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Created by sakshia on 26/10/16.
+ */
+
+public class LocalTransferServer
+{
+    private static Log log = LogFactory.getLog(LocalTransferServer.class.getName());
+    private static Configuration conf;
+    private static  LocalServer localServer;
+
+    private LocalTransferServer()
+    {
+    }
+
+    public static void main(String[] args)
+    {
+        conf = new Configuration();
+        startServer(conf);
+    }
+
+    public static void startServer(Configuration conf)
+    {
+        localServer = new LocalServer(conf);
+        new Thread(localServer).run();
+    }
+
+    public static void stopServer()
+    {
+        localServer.stop();
+    }
+
+    public static class LocalServer implements Runnable
+    {
+        static ServerSocketChannel listener;
+        Configuration conf;
+
+        public LocalServer(Configuration conf)
+        {
+            this.conf = conf;
+        }
+
+        @Override
+        public void run()
+        {
+            int port = CacheConfig.getLocalServerPort(conf);
+
+            ExecutorService threadPool = Executors.newCachedThreadPool();
+            try {
+                listener = ServerSocketChannel.open();
+                listener.bind(new InetSocketAddress(port));
+                log.info("Listening on port " + port);
+                while (true) {
+                    SocketChannel clientSocket = listener.accept();
+                    ClientServiceThread cliThread = new ClientServiceThread(clientSocket);
+                    threadPool.execute(cliThread);
+                }
+            }
+            catch (AsynchronousCloseException e) {
+                log.info("Stopping Local Transfer server");
+            }
+            catch (IOException e) {
+                log.error(String.format("Error starting Local Transfer server %s", Throwables.getStackTraceAsString(e)));
+                e.printStackTrace();
+            }
+
+        }
+
+        public void stop()
+        {
+            try {
+                listener.close();
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+    static class ClientServiceThread
+            extends Thread
+    {
+        SocketChannel localTransferClient;
+        RetryingBookkeeperClient bookKeeperClient;
+
+        ClientServiceThread(SocketChannel s)
+        {
+            localTransferClient = s;
+        }
+
+        public void run()
+        {
+            try {
+                log.debug("Connected to node - " + localTransferClient.getLocalAddress());
+                BookKeeperFactory bookKeeperFactory = new BookKeeperFactory();
+                ByteBuffer dataInfo = ByteBuffer.allocate(CacheConfig.getDataTransferBufferSize(conf));
```

Comment:
is 1024 bytes enough? There can be very long filenames since they are absolute paths